app/components/month-selector.js

```diff
@@ -0,0 +1,65 @@
+import Component from '@glimmer/component';
+import { sub, add, format } from 'date-fns';
+import { monthsInYear } from 'date-fns/constants';
+import { service } from '@ember/service';
+import { localCopy } from 'tracked-toolbox';
+import { trackedFunction } from 'reactiveweb/function';
+
+export default class MonthSelectComponent extends Component {
+  @service('timesheets') timesheetsService;
+  @localCopy('args.year') year;
+
+  timesheets = trackedFunction(this, async () => {
+    return await this.timesheetsService.getForYear(this.year);
+  });
+
+  decrementYear = () => {
+    this.year -= 1;
+  };
+
+  incrementYear = () => {
+    this.year += 1;
+  };
+
+  get months() {
+    const items = [];
+    for (let i = 0; i < monthsInYear; i++) {
+      const dateInMonth = new Date(this.args.year, i, 1);
+      items.push({
+        number: i,
+        humanNumber: i + 1,
+        label: format(dateInMonth, 'LLLL'),
+        timesheet: this.timesheets.value?.find(
+          (timesheet) => timesheet.start.getMonth() === i,
+        ),
```

Comment:
`this.timesheets` only contains the timesheets queried from the backend and not the draft ones created in the model-hook of the `year` route. Hence, `timesheet` will be `null` for those. That explains the issue you encountered where the status is not being updated in the dropdown.

I would update the TimesheetsService and make `getForYear` responsible to ensure a timesheet for each month (thus, including creation of draft timesheets). I expect this will resolve the bug.